### ISSUE TITLE
Polkit details support

### ISSUE THREAD

**Issue by ludgie** [OmarCastro/cmd-polkit#8]
Hello!

I've read the docs, but does this library support parsing details (ActionDescription struct) or only prompting?
More info: https://www.freedesktop.org/software/polkit/docs/latest/eggdbus-interface-org.freedesktop.PolicyKit1.Authority.html

Thanks!

**Comment by OmarCastro** [comment]
Hello, 

The idea of this application is to simplify the creation of UI application from prompting, what is the purpose of using the ActionDescription? What you receive is the message used to show in the authentication dialog

**Comment by ludgie** [comment]
Got it, thank you!

**Comment by ludgie** [comment]
Reopening


> Hello,
> 
> The idea of this application is to simplify the creation of UI application from prompting, what is the purpose of using the ActionDescription? What you receive is the message used to show in the authentication dialog

Well, maybe the user that is using cmd-polkit to customize the polkit UI wants to see as well ActionDescription?
![image](https://github.com/OmarCastro/cmd-polkit/assets/77110723/52d0ef01-8691-44ae-b328-82092060672a)

**Comment by OmarCastro** [comment]
Ok, I added a field on the message on the documentation, the field name is `action description`, but it is not final. Given there is the field `action`, it can give the wrong impression that they are related.

I am going to use this issue to discuss the name. I am thinking to change the name to `polkit action description`, but not only it is long, but also this project is specific to polkit. I also thought about `action details`, however it is not future proof there is [PolkitDetails](https://www.freedesktop.org/software/polkit/docs/latest/PolkitDetails.html) that most likely will be requested to add on the password request message

What do you think @ludg1e ?

**Comment by ludgie** [comment]
Hello,

What about something like the following?
```
{
              "action": "request password",
              "prompt": string 
              "message": string 
              "requester details": null | {
                "id": string
                "description": string
                "message": string
                "vendor name": string
                "vendor url": string
                "icon name": string
              }
```

`requester` makes reference to the program that is asking for privileges, in this case through polkit.

Regarding PolkitDetails, per the documentation I've found this
https://lazka.github.io/pgi-docs/Polkit-1.0/classes/Authority.html#Polkit.Authority.check_authorization

```
Known keys in details include polkit.message and polkit.gettext_domain that can be used to override the message shown to the user. See the documentation for the ‘D-Bus method [eggdbus-method-org.freedesktop.PolicyKit1.Authority.CheckAuthorization]’ for more details.

If details is non-empty then the request will fail with [Polkit.Error.FAILED](https://lazka.github.io/pgi-docs/Polkit-1.0/enums.html#Polkit.Error.FAILED) unless the process doing the check itsef is sufficiently authorized (e.g. running as uid 0).
```

**Comment by OmarCastro** [comment]
I studied about it, and I came on the conclusion that the best name is `polkit action`, according to the documentation in https://www.freedesktop.org/software/polkit/docs/latest/polkit.8.html there is this part in "Declaring actions" section:

![imagem](https://github.com/OmarCastro/cmd-polkit/assets/1904134/045669aa-1fe4-4dd5-a12f-b46cf135f76c)

I highlighted the first letters of the second paragraph, they call "polkit actions", and it uses the same model. This way, "polkit action" will mean the "Polkit Action" entity related to the program that is asking for privilege, and the meaning is consistent with Polkit documentation.

Meanwhile, I will reserve `details` to PolkitDetails, if ever needed, since details mean "Details describing the action" here: https://github.com/polkit-org/polkit/blob/e89a5db29a18bb4091d1f3ae5d1e1f8f98b63bb7/src/polkitagent/polkitagentlistener.c#L743

While `requester details` make sense, it is a bit confusing since the idea that I got is that we are referencing the application requesting the authentication. So, by having different `id`s we will reference different application which is incorrect, since one application can register multiple actions. Also, the description and message are describes the action instead of the program
